db/builder.cc

```diff
@@ -30,14 +30,12 @@ Status BuildTable(const std::string& dbname, Env* env, const Options& options,
 
     TableBuilder* builder = new TableBuilder(options, file);
     meta->smallest.DecodeFrom(iter->key());
-    Slice key;
     for (; iter->Valid(); iter->Next()) {
-      key = iter->key();
-      builder->Add(key, iter->value());
-    }
-    if (!key.empty()) {
-      meta->largest.DecodeFrom(key);
+      builder->Add(iter->key(), iter->value());
     }
+    
+    iter->SeekToLast();
+    meta->largest.DecodeFrom(iter->key());
```

Comment:
Must check if `iter->Valid()` between the `SeekToLast` and `iter->key()` call.

Reply:
> Must check if `iter->Valid()` between the `SeekToLast` and `iter->key()` call.

Good point - will fix it!